src/components/Button.js

```diff
@@ -1,10 +1,17 @@
 import styled from 'styled-components';
+import {Icon} from '@iconify/react';
 
-export const Button = ({textValue}) => {
-  return <StyledButton type="submit">{textValue}</StyledButton>;
+export const Button = ({textValue, onClick, dataid}) => {
+  return (
+    <StyledButton type="submit" onClick={onClick} data-id={dataid}>
+      {textValue === 'Delete' ? '' : textValue}
+      {textValue === 'Delete' ? <Icon icon="ant-design:delete-outlined" style={{fontsize: '28px'}} /> : ''}
```

Comment:
You do not need two different tenery's for this case. You can solve this with one single tenery.